outsystems/outsystems-quirks.md

```diff
@@ -2,5 +2,12 @@
 
 This is an area where we can record any strange/unexpected things that outsystems does so we can avoid them effecting us in future projects
 
-###Issue with renaming table columns
-When deploying to the next lifetime environment and a table column has been renamed, OutSystems will not rename the column in the next environment. It will create a new column and it will not copy across the data from the original column.
\ No newline at end of file
+## Issue with renaming table columns
+
+When deploying to the next lifetime environment and a table column has been renamed, OutSystems will not rename the column in the next environment. It will create a new column and it will not copy across the data from the original column.
+
+The reason for this is that if you needed to roll back the application it would cause all sorts of data issues.
+
+### 2 possible solutions for this are
+1. Don't rename the column just create a new one of the same data type with the new name you want and change where all the values are assigned and viewed within your application to use that new column. Then set up a timer job to copy across all values from the old version into the new one. The important thing to remember for this one is that you turn the timer job off after the first publish otherwise it'll delete the old data. 
+2. Rename the column as normal and once you have published your application log into the sql server database and copy the data from the old column into the new one. The risks of this is that because you are directly on the database you have more power with your changes and could more easily make breaking changes within your data.
```

Comment:
Is it worth _only_ suggesting this option as the advised method?

Is the risk of someone accidentally running a timer more than once or leaving a scheduled timer running greater than the risk of someone accessing the database directly?